src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -35,6 +35,7 @@ private enum ParameterSyntaxKind : byte
         private ThreeState _lazyHasOptionalAttribute;
         private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
         protected ConstantValue _lazyDefaultSyntaxValue;
+        protected BoundParameterEqualsValue _lazyBoundEqualsValueSyntax = null;
```

Comment:
It can be problematic to include bound nodes in symbols because it causes the symbols to retain references to syntax nodes which could otherwise be GCed. This is most applicable for member symbols.

This field may only be needed for local symbols, which I *think* can reference syntax without problems. It might be possible to adjust the conditions when we store a non-null value here to mitigate the problem. For example, maybe we could ensure we only store a non-null value here if the containing symbol is a lambda or local function.

However, it might be cleaner to modify `BoundLocalFunctionStatement`/`BoundLambda` instead to store these values in the bound tree. I'm not sure how disruptive of a change that would be, since currently the default value seems to be bound on-demand separately from the method body.